### PR TITLE
feat(app): update and use error page

### PR DESCRIPTION
**Pull request by marie-j**
## update and use error page

MANAGER-3136

### Description of the Change

* Refactor error page into a component 
* Trigger error page on resolve failure

### CODE REVIEW

File: client/app/app.js [hunk 1]
```diff
@@ -154,7 +156,7 @@ angular
   .run((ssoAuthentication, User) => {
     ssoAuthentication.login().then(() => User.getUser());
   })
-  .run(($transitions, $rootScope, $state, coreConfig) => {
+  .run(/* @ngInject */ ($transitions, $rootScope, $state, coreConfig) => {
```
Comment (JeremyDec) [inline client/app/app.js]:
```suggestion
  .run(/* @ngInject */ ($rootScope, $state, $transitions, coreConfig) => {
```

File: client/app/app.js [hunk 2]
```diff
@@ -19,11 +19,12 @@ import ovhManagerCore from '@ovh-ux/manager-core';
 import ovhManagerNavbar from '@ovh-ux/manager-navbar';
 import ovhManagerServerSidebar from '@ovh-ux/manager-server-sidebar';
 import ovhPaymentMethod from '@ovh-ux/ng-ovh-payment-method';
-import uiRouter from '@uirouter/angularjs';
+import uiRouter, { RejectType } from '@uirouter/angularjs';
```
Comment (JeremyDec) [inline client/app/app.js]: Nice :)

File: client/app/error/error.routing.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('app.error', {
+    url: '/error',
+    params: {
+      detail: null,
+    },
+    views: {
+      'app@': {
```
Comment (JeremyDec) [inline client/app/error/error.routing.js]: Is there a reason to not specify the state where the `app` view is located ?

Comment (marie-j) [reply client/app/error/error.routing.js]: The `app` ui-view doesn't seem associated with any states

File: client/app/error/translations/Messages_fr_FR.json [hunk 4]
```diff
@@ -2,5 +2,7 @@
   "page_load_error": "Une erreur est survenue lors du chargement de la page.",
   "page_load_error_common": "La page demandée n'a pas été trouvée.",
   "page_load_error_forbidden_billing_access": "La section \"Facturation\" n'est pas accessible depuis ce compte.",
-  "page_load_error_back_to_home": "Retour à l'accueil de votre espace client"
-}
\ No newline at end of file
+  "page_load_error_back_to_home": "Retour à la page d'accueil",
+  "page_load_error_code": "Code d'erreur: {{ code }}",
```
Comment (antleblanc) [inline client/app/error/translations/Messages_fr_FR.json]:
```suggestion
  "page_load_error_code": "Code d'erreur : {{ code }}",
```

**Pull request merged**